src/tests/roc_audio/test_depacketizer.cpp

```diff
@@ -189,6 +189,73 @@ TEST(depacketizer, drop_late_packets) {
     expect_output(dp, SamplesPerPacket, 0.33f);
 }
 
+TEST(depacketizer, frame_has_dropped_packets_at_the_end) {
+    packet::ConcurrentQueue queue(0, false);
+    Depacketizer dp(queue, pcm_decoder, ChMask, false);
+
+    const packet::timestamp_t ts1 = SamplesPerPacket * 3;
+    const packet::timestamp_t ts2 = SamplesPerPacket * 2;
+    const packet::timestamp_t ts3 = SamplesPerPacket * 1;
+
+    queue.write(new_packet(ts1, 0.11f));
+    queue.write(new_packet(ts2, 0.22f));
+    queue.write(new_packet(ts3, 0.33f));
+
+    Frame frame = new_frame(SamplesPerPacket * 3);
+    dp.read(frame);
+
+    CHECK(frame.flags() & Frame::FlagSkip);
+}
+
+TEST(depacketizer, frame_has_dropped_packets_in_the_middle) {
+    packet::ConcurrentQueue queue(0, false);
+    Depacketizer dp(queue, pcm_decoder, ChMask, false);
+
+    const packet::timestamp_t ts1 = SamplesPerPacket * 3;
+    const packet::timestamp_t ts2 = SamplesPerPacket * 2;
+    const packet::timestamp_t ts3 = SamplesPerPacket * 4;
+
+    queue.write(new_packet(ts1, 0.11f));
+    queue.write(new_packet(ts2, 0.22f));
+    queue.write(new_packet(ts3, 0.33f));
+
+    Frame frame = new_frame(SamplesPerPacket * 3);
+    dp.read(frame);
+
+    CHECK(frame.flags() & Frame::FlagSkip);
+}
+
+TEST(depacketizer, frame_has_no_dropped_packets) {
+    packet::ConcurrentQueue queue(0, false);
+    Depacketizer dp(queue, pcm_decoder, ChMask, false);
+
+    const packet::timestamp_t ts1 = SamplesPerPacket;
+    const packet::timestamp_t ts2 = SamplesPerPacket * 2;
+
+    queue.write(new_packet(ts1, 0.11f));
+    queue.write(new_packet(ts2, 0.22f));
+
+    Frame frame = new_frame(SamplesPerPacket * 2);
+    dp.read(frame);
+
+    CHECK(!(frame.flags() & Frame::FlagSkip));
+}
+
+TEST(depacketizer, empty_frame) {
```

Comment:
Could you also add a test for the case when Frame::FlagEmpty is *not* set?

Reply:
I'll expand the current test